### PR TITLE
Game stores2

### PR DESCRIPTION
**Pull request by stanco2rhyme**
Resolve Merge conflict
Change initial entry point for VC

### CODE REVIEW

File: BoardBank/MonoBank/AddPlayerViewController.swift [hunk 1]
```diff
@@ -5,6 +5,7 @@
 //  Created by Richard Neitzke on 04/01/2017.
 //  Copyright © 2017 Richard Neitzke. All rights reserved.
 //
+// swiftlint:disable all
```
Comment (prrenfrew) [inline BoardBank/MonoBank/AddPlayerViewController.swift]: Remove this line.

File: BoardBank/MonoBank/BankManager.swift [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  /// Saves the current state of the BankManager
+  func save() {
+    UserDefaults.standard.set(currencySymbol, forKey: "currencySymbol")
+    UserDefaults.standard.set(defaultBalance, forKey: "defaultBalance")
+    UserDefaults.standard.set(quickAddAmount, forKey: "quickAddAmount")
+    let playersData = NSKeyedArchiver.archivedData(withRootObject: players)
+    UserDefaults.standard.set(playersData, forKey: "players")
+    UserDefaults.standard.set(soundsEnabled, forKey: "soundsEnabled")
+  }
+}
```
Comment (prrenfrew) [inline BoardBank/MonoBank/BankManager.swift]: Why was this added?

File: BoardBank/MonoBank/GameStores.swift [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    init(mapItem: MKMapItem) {
+        self.title = mapItem.placemark.name
+        self.locationName = mapItem.placemark.name
+        self.address = mapItem.placemark.title
+        self.phoneNumber = mapItem.phoneNumber
+        self.coordinate = mapItem.placemark.coordinate
+    }
+
+    var subtitle: String? {
```
Comment (prrenfrew) [inline BoardBank/MonoBank/GameStores.swift]: What is the purpose of this computed property?

File: BoardBank/MonoBank/MainViewController.swift [hunk 4]
```diff
@@ -6,6 +6,8 @@
 //  Copyright © 2017 Richard Neitzke. All rights reserved.
 //
 
+// swiftlint:disable all
```
Comment (prrenfrew) [inline BoardBank/MonoBank/MainViewController.swift]: Remove this line

File: BoardBank/MonoBank/MapTableViewController.swift [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+//  BoardBank
+//
+//  Created by Stanley Ejechi on 4/30/19.
+//  Copyright © 2019 Richard Neitzke. All rights reserved.
+
+import UIKit
+import CoreLocation
+import MapKit
+
+class MapTableViewController: UIViewController {
```
Comment (prrenfrew) [inline BoardBank/MonoBank/MapTableViewController.swift]: This view controller isn't doing anything. It has a table view, but it doesn't set up the datasource or delegate, and the protocols for them are not implemented.

File: BoardBank/MonoBank/MapViewController.swift [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+        manager.delegate = self
+        manager.startUpdatingLocation()
+    }
+    // MARK: Perform search for nearby store
+    func queryStores(region: MKCoordinateRegion) {
+        let request = MKLocalSearch.Request()
+        request.naturalLanguageQuery = "Game Stores"
+        request.region = region
+        let search = MKLocalSearch(request: request)
+        search.start(completionHandler: {(response, error) in
```
Comment (prrenfrew) [inline BoardBank/MonoBank/MapViewController.swift]: Use trailing closure syntax

Comment (prrenfrew) [reply BoardBank/MonoBank/MapViewController.swift]: Also, you need a space between opening braces and the opening parenthesis.

File: BoardBank/MonoBank/MapViewController.swift [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+            guard let count = response?.mapItems.count else {return}
+            guard let responseMapItems = response?.mapItems else {return}
```
Comment (prrenfrew) [inline BoardBank/MonoBank/MapViewController.swift]: Spaces between braces and return.

File: BoardBank/MonoBank/MapViewController.swift [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+            if error != nil {
+            } else if count == 0 {
```
Comment (prrenfrew) [inline BoardBank/MonoBank/MapViewController.swift]: Why are we checking this conditions if we don't do anything if they are true?

File: BoardBank/MonoBank/MapViewController.swift [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+        let annotation = MKPointAnnotation()
+        annotation.coordinate = location.coordinate
+        theMapView.addAnnotation(annotation)
+        theMapView.mapType = .standard
+        theMapView.delegate = self
+        queryStores(region: coordinateRegion)
+    }
+
+    // MARK: Table Button Action Method
+    @IBAction func tableButtonPressed(_ sender: Any) {
```
Comment (prrenfrew) [inline BoardBank/MonoBank/MapViewController.swift]: Why does this IBAction do nothing? This results in a bad user experience when a button doesn't do anything when tapped.

File: BoardBank/MonoBank/MapViewController.swift [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+
+    func errorAlert(description: String) {
+        let alert = UIAlertController(title: "Error",
+                                      message: description,
+                                      preferredStyle: UIAlertController.Style.alert)
+
+        alert.addAction(UIAlertAction(title: "OK",
+                                      style: UIAlertAction.Style.default,
+                                      handler: { _ in
+        }))
```
Comment (prrenfrew) [inline BoardBank/MonoBank/MapViewController.swift]: The handler is optional. Since you are doing nothing, you should pass in nil as opposed to an empty closure.

File: BoardBank/MonoBank/MapViewController.swift [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+    func mapView(_ mapView: MKMapView, viewFor annotation: MKAnnotation) -> MKAnnotationView? {
+            guard let annotation = annotation as? GameStore else { return nil }
+            let identifier = "marker"
+            var view: MKMarkerAnnotationView
+            if let dequeuedView = mapView.dequeueReusableAnnotationView(withIdentifier: identifier)
+                as? MKMarkerAnnotationView {
+                dequeuedView.annotation = annotation
+                view = dequeuedView
+                view.glyphImage = UIImage(named: "store")
+                view.selectedGlyphImage = UIImage(named: "store")} else {
```
Comment (prrenfrew) [inline BoardBank/MonoBank/MapViewController.swift]: The brace and else statement need to go on a separate line.

File: BoardBank/MonoBank/MapViewController.swift [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+                view = MKMarkerAnnotationView(annotation: annotation, reuseIdentifier: identifier)
+                view.canShowCallout = true
+                view.calloutOffset = CGPoint(x: -5, y: 5)
```
Comment (prrenfrew) [inline BoardBank/MonoBank/MapViewController.swift]: When you have hardcoded numbers like this, they are referred to as "magic numbers". Basically, numbers that we don't know why they are used. Refrain from using magic numbers.

File: BoardBank/MonoBank/MapViewController.swift [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+                view.rightCalloutAccessoryView = UIButton(type: .detailDisclosure)
+                view.glyphImage = UIImage(named: "store")
+                view.selectedGlyphImage = UIImage(named: "store")
+            }
+            return view
+    }
+
+    func mapView(_ mapView: MKMapView, annotationView view: MKAnnotationView,
+                 calloutAccessoryControlTapped control: UIControl) {
+        if let location = view.annotation as? GameStore {
```
Comment (prrenfrew) [inline BoardBank/MonoBank/MapViewController.swift]: Since this function doesn't do anything if the if statement condition fails, use a guard statement for cleaner code.

File: BoardBank/MonoBank/MapViewController.swift [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+        }
+    }
+}
+// MARK: CLLocationManagerDelegate Method
+extension MapViewController: CLLocationManagerDelegate {
+
+    func locationManager(_ manager: CLLocationManager,
+                         didUpdateLocations locations: [CLLocation]) {
+        guard let currentLocation = locations.last else { return }
+        self.manager.stopUpdatingLocation()
```
Comment (prrenfrew) [inline BoardBank/MonoBank/MapViewController.swift]: Only use self when you need to, such as within a closure.

File: BoardBank/MonoBank/Model/BankManager.swift [hunk 15]
```diff
@@ -5,6 +5,7 @@
 //  Created by Richard Neitzke on 03/01/2017.
 //  Copyright © 2017 Richard Neitzke. All rights reserved.
 //
+// swiftlint:disable all
```
Comment (prrenfrew) [inline BoardBank/MonoBank/Model/BankManager.swift]: Remove this line.

File: BoardBank/MonoBank/SettingsViewController.swift [hunk 16]
```diff
@@ -5,6 +5,7 @@
 //  Created by Richard Neitzke on 04/01/2017.
 //  Copyright © 2017 Richard Neitzke. All rights reserved.
 //
+// swiftlint:disable all
```
Comment (prrenfrew) [inline BoardBank/MonoBank/SettingsViewController.swift]: Remove this line.

File: BoardBank/MonoBank/GameStores.swift [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+//  BoardBank
+//
+//  Created by Stanley Ejechi on 5/1/19.
+//  Copyright © 2019 Richard Neitzke. All rights reserved.
+
+import Foundation
+import MapKit
+import Contacts
+
+class GameStore: NSObject, MKAnnotation {
```
Comment (prrenfrew) [inline BoardBank/MonoBank/GameStores.swift]: This class name does not tell us that it is an annotation, so the name is not descriptive enough.

**Comment by prrenfrew** [comment]
@stanco2rhyme left some comments. Please reopen once all comments are addressed.

File: BoardBank/MonoBank/AddPlayerViewController.swift (hunk 1 above)
Comment (stanco2rhyme) [reply BoardBank/MonoBank/AddPlayerViewController.swift]: ok

File: BoardBank/MonoBank/BankManager.swift (hunk 2 above)
Comment (stanco2rhyme) [reply BoardBank/MonoBank/BankManager.swift]: I didn't add this. Not my code

File: BoardBank/MonoBank/GameStores.swift (hunk 3 above)
Comment (stanco2rhyme) [reply BoardBank/MonoBank/GameStores.swift]: i removed it. Not really needed

File: BoardBank/MonoBank/MapTableViewController.swift (hunk 5 above)
Comment (stanco2rhyme) [reply BoardBank/MonoBank/MapTableViewController.swift]: removed it

File: BoardBank/MonoBank/MapViewController.swift (hunk 6 above)
Comment (stanco2rhyme) [reply BoardBank/MonoBank/MapViewController.swift]: ok

File: BoardBank/MonoBank/MapViewController.swift (hunk 7 above)
Comment (stanco2rhyme) [reply BoardBank/MonoBank/MapViewController.swift]: ok

File: BoardBank/MonoBank/MapViewController.swift (hunk 8 above)
Comment (stanco2rhyme) [reply BoardBank/MonoBank/MapViewController.swift]: fix it

File: BoardBank/MonoBank/MapViewController.swift (hunk 9 above)
Comment (stanco2rhyme) [reply BoardBank/MonoBank/MapViewController.swift]: removed it. I wanted to add more functionality

File: BoardBank/MonoBank/MapViewController.swift (hunk 10 above)
Comment (stanco2rhyme) [reply BoardBank/MonoBank/MapViewController.swift]: fixed

File: BoardBank/MonoBank/GameStores.swift (hunk 3 above)
Comment (stanco2rhyme) [reply BoardBank/MonoBank/GameStores.swift]: I added it back incase i decide to change my substring to return something else

File: BoardBank/MonoBank/MapViewController.swift (hunk 12 above)
Comment (stanco2rhyme) [reply BoardBank/MonoBank/MapViewController.swift]: Okay. Not needed

File: BoardBank/MonoBank/MapViewController.swift (hunk 13 above)
Comment (stanco2rhyme) [reply BoardBank/MonoBank/MapViewController.swift]: ok

File: BoardBank/MonoBank/MapViewController.swift (hunk 14 above)
Comment (stanco2rhyme) [reply BoardBank/MonoBank/MapViewController.swift]: ok

File: BoardBank/MonoBank/Model/BankManager.swift (hunk 15 above)
Comment (stanco2rhyme) [reply BoardBank/MonoBank/Model/BankManager.swift]: ok

File: BoardBank/MonoBank/SettingsViewController.swift (hunk 16 above)
Comment (stanco2rhyme) [reply BoardBank/MonoBank/SettingsViewController.swift]: ok

File: BoardBank/MonoBank/GameStores.swift (hunk 17 above)
Comment (stanco2rhyme) [reply BoardBank/MonoBank/GameStores.swift]: ok
